haystack/agents/conversational.py

```diff
@@ -1,63 +1,122 @@
+        "transcript": agent_step.transcript,
+        "history": agent.memory.load(),
+    }
 
 
 class ConversationalAgent(Agent):
     """
-    A conversational agent that can be used to build a conversational chat applications.
+    A ConversationalAgent is an extension of the Agent class that enables the use of tools with several default parameters.
+    ConversationalAgent can manage a set of tools and seamlessly integrate them into the conversation.
+    If no tools are provided, the agent will be initialized to have a basic chat application.
 
-    Here is an example of how you can create a simple chat application:
-    ```
+    Here is an example how you can create a chat application with tools:
+    ```python
     import os
 
-    from haystack.agents.base import ConversationalAgent
-    from haystack.agents.memory import ConversationSummaryMemory
+    from haystack.agents.conversational import ConversationalAgent
     from haystack.nodes import PromptNode
+    from haystack.agents.base import ToolsManager, Tool
+
+    # Initialize a PromptNode and a ToolsManager with the desired tools
+    prompt_node = PromptNode("gpt-3.5-turbo", api_key=os.environ.get("OPENAI_API_KEY"), max_length=256)
+    tools = [Tool(name="ExampleTool", pipeline_or_node=example_tool_node)]
 
-    pn = PromptNode("gpt-3.5-turbo", api_key=os.environ.get("OPENAI_API_KEY"), max_length=256)
-    agent = ConversationalAgent(pn, memory=ConversationSummaryMemory(pn))
+    # Create the ConversationalAgent instance
+    agent = ConversationalAgent(prompt_node, tools=tools)
 
+    # Use the agent in a chat application
     while True:
         user_input = input("Human (type 'exit' or 'quit' to quit): ")
         if user_input.lower() == "exit" or user_input.lower() == "quit":
             break
-        elif user_input.lower() == "memory":
-            print("\nMemory:\n", agent.memory.load())
         else:
             assistant_response = agent.run(user_input)
             print("\nAssistant:", assistant_response)
+    ```
 
+    If you don't want to have any tools in your chat app, you can create a ConversationalAgent only with a PromptNode:
+    ```python
+    import os
+
+    from haystack.agents.conversational import ConversationalAgent
+    from haystack.nodes import PromptNode
+
+    # Initialize a PromptNode
+    prompt_node = PromptNode("gpt-3.5-turbo", api_key=os.environ.get("OPENAI_API_KEY"), max_length=256)
+
+    # Create the ConversationalAgent instance
+    agent = ConversationalAgent(prompt_node)
     ```
+
+    If you're looking for more customization, check out [Agent](https://docs.haystack.deepset.ai/reference/agent-api).
     """
 
-    def __init__(
-        self,
-        prompt_node: PromptNode,
-        prompt_template: Optional[Union[str, PromptTemplate]] = None,
-        memory: Optional[Memory] = None,
-        prompt_parameters_resolver: Optional[Callable] = None,
```

Comment:
Why not allow users to set this parameter prompt_parameters_resolver and have None as default? Similar for max_steps, final_answer_pattern and streaming: Why not allow the user to set these and at the same time have defaults.
For example, with the current implementation max_steps is set to 2 and can't be changed if there are no tools

Reply:
IMO, if we enable setting these parameters (final_answer_patter) etc, through ConversationalAgent, then it can be easily used instead of Agent. Do we want this? If we see that users want to customize these params, then we can guide them to use Agent rather than ConversationalAgent

Reply:
I better understand this reasoning now, thank you for further explaining it. 🙏 What I have in mind is that a user who wants to use a ConversationalAgent but with max_steps=3 can't use ConversationalAgent. Instead they will need to initialize an Agent not only with max_steps=3 but also with setting prompt_template, final_answer_pattern, prompt_parameters_resolver and memory (manually copying over the settings from ConversationalAgent).
We need to decide on a good balance of flexibility and simplicity here. Would be curious to hear your opinions @bogdankostic @masci 🙂

Reply:
Good point about `max_steps` @julian-risch 👍 We can definitely add this param to ConversationalAgent.